c_glib/arrow-glib/type.h

```diff
@@ -94,6 +95,7 @@ typedef enum {
   GARROW_TYPE_TIME64,
   GARROW_TYPE_INTERVAL_MONTHS,
   GARROW_TYPE_INTERVAL_DAY_TIME,
+  GARROW_TYPE_INTERVAL_MONTH_DAY_NANO,
```

Comment:
We should not put this here.
We need to use the same order as `arrow::Type::type` to use the same integer value.